src/context/product-context.js

```diff
@@ -0,0 +1,51 @@
+import { createContext, useContext, useReducer } from 'react';
+
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+  const productReducerFunc = (state, action) => {
+    switch (action.type) {
+      case 'FETCH_PRODUCT':
+        return { ...state, filteredProducts: action.payload };
+      case 'ADD_TO_CART':
+        return {
+          ...state,
+          cartProducts: action.payload,
+        };
+      case 'REMOVE_FROM_CART':
+        return { ...state, cartProducts: action.payload };
+
+      case 'ADD_TO_WISHLIST':
+        return {
+          ...state,
+          wishListProducts: action.payload,
+          wishListed: !state.wishListed,
+        };
+      case 'REMOVE_FROM_WISHLIST':
+        return {
+          ...state,
+          wishListProducts: action.payload,
+          wishListed: !state.wishListed,
```

Comment:
For a better codebase design, I separated wishlist and cart functionality in different files but I guess this is also fine. Plus if you make this change now, you will have to update at a lot of places.